packages/cli/src/commands/storybook.js

```diff
@@ -43,24 +47,30 @@ export const builder = (yargs) => {
       type: 'boolean',
       default: false,
     })
-    .check((argv) => {
-      if (argv.build && argv.smokeTest) {
-        throw new Error('Can not provide both "--build" and "--smoke-test"')
-      }
-      if (argv.build && argv.open) {
-        throw new Error('Can not provide both "--build" or "--open"')
-      }
-      return true
-    })
+    // .check((argv) => {
+    //   if (argv.build && argv.smokeTest) {
+    //     throw new Error('Can not provide both "--build" and "--smoke-test"')
+    //   }
+    //   if (argv.build && argv.open) {
+    //     throw new Error('Can not provide both "--build" or "--open"')
+    //   }
+    //   return true
+    // })
```

Comment:
Even though these combinations will no longer cause the command to fail, I still think it's good to provide this feedback. 

For --build=true and --smoke-test=true, I think we have to throw.

For --build=true and --open=true, we could console output instead of throwing. E.g. something like "Warning: --open option has no effect when running Storybook build"